modules/schematics/src/create-action-migration/index.ts

```diff
@@ -0,0 +1,206 @@
+import * as ts from 'typescript';
+import { Path, tags } from '@angular-devkit/core';
+import { camelize } from '@angular-devkit/core/src/utils/strings';
+import { Tree, Rule, chain } from '@angular-devkit/schematics';
+import {
+  InsertChange,
+  RemoveChange,
+  ReplaceChange,
+  replaceImport,
+  commitChanges,
+  Change,
+} from '@ngrx/schematics/schematics-core';
+
+export function migrateToCreators(): Rule {
+  return (host: Tree) =>
+    host.visit(path => {
+      if (!path.endsWith('.ts') || path.includes('node_modules')) {
+        return;
+      }
+
+      const sourceFile = ts.createSourceFile(
+        path,
+        host.read(path)!.toString(),
+        ts.ScriptTarget.Latest
+      );
+
+      if (sourceFile.isDeclarationFile || !isActionImported(sourceFile)) {
+        return;
+      }
+
+      const actionClasses = getActionClasses(sourceFile);
+      const actionTypeVariables = sourceFile.statements.filter(
+        ts.isVariableStatement
+      );
+
+      let importProps = false;
+
+      let changes = (actionClasses
+        .map(actionClass => {
+          let typeProperty = actionClass.members
+            .filter(ts.isPropertyDeclaration)
+            .filter(member => (member.name as ts.Identifier).text === 'type')
+            .map(member => (member.initializer as ts.Identifier).text)[0];
+
+          if (!typeProperty) {
+            return undefined;
+          }
+
+          const typeVariable = actionTypeVariables.filter(
+            statement =>
+              statement.declarationList.declarations.filter(
+                node => (node.name as ts.Identifier).text === typeProperty
+              ).length
+          );
+
+          const typeValue = typeVariable
+            .map(statement => statement.declarationList.declarations)
+            .map(decl => (decl[0].initializer as ts.Identifier).text)[0];
+
+          if (!typeValue) {
+            return undefined;
+          }
+
+          const payloadType = getConstructorPayloadType(
+            host,
+            path,
+            actionClass
+          );
+          if (payloadType) {
+            importProps = true;
```

Comment:
Would this cause for problems, we set `importProps` to `false` outside of the loop.
But we never re-assign it back to `false`, thus once it's assigned to `true` it will always be `true`.